src/hotspot/share/opto/superword.cpp

```diff
@@ -1996,291 +1961,20 @@ void PackSet::verify() const {
-  // note if topsort was successful.
-  Node_List schedule() {
-    Node_List memops_schedule;
-    GrowableArray<int> worklist;
-    // Directly schedule all nodes without precedence
-    for (int pid = 1; pid <= _max_pid; pid++) {
-      if (incnt(pid) == 0) {
-        worklist.push(pid);
-      }
-    }
-    // Continue scheduling via topological sort
-    for (int i = 0; i < worklist.length(); i++) {
-      int pid = worklist.at(i);
-
-      // Add memops to memops_schedule
-      Node* n = get_node(pid);
-      Node_List* p = _slp->packset().get_pack(n);
-      if (n->is_Mem()) {
-        if (p == nullptr) {
-          memops_schedule.push(n);
-        } else {
-          for (uint k = 0; k < p->size(); k++) {
-            memops_schedule.push(p->at(k));
-            assert(p->at(k)->is_Mem(), "only schedule memops");
-          }
-        }
-      }
-
-      // Decrement incnt for all successors
-      for (int j = 0; j < out(pid).length(); j++){
-        int pid_use = out(pid).at(j);
-        int incnt_use = incnt(pid_use) - 1;
-        incnt_set(pid_use, incnt_use);
-        // Did use lose its last input?
-        if (incnt_use == 0) {
-          worklist.push(pid_use);
-        }
-      }
-    }
-
-    // Was every pid scheduled? If not, we found some cycles in the PacksetGraph.
-    _schedule_success = (worklist.length() == _max_pid);
-    return memops_schedule;
-  }
-
-  // Print the PacksetGraph.
-  // print_nodes = true: print all C2 nodes beloning to PacksetGrahp node.
-  // print_zero_incnt = false: do not print nodes that have no in-edges (any more).
-  void print(bool print_nodes, bool print_zero_incnt) {
-    const GrowableArray<Node*> &body = _slp->body();
-    tty->print_cr("PacksetGraph");
-    for (int pid = 1; pid <= _max_pid; pid++) {
-      if (incnt(pid) == 0 && !print_zero_incnt) {
-        continue;
-      }
-      tty->print("Node %d. incnt %d [", pid, incnt(pid));
-      for (int j = 0; j < out(pid).length(); j++) {
-        tty->print("%d ", out(pid).at(j));
-      }
-      tty->print_cr("]");
-#ifndef PRODUCT
-      if (print_nodes) {
-        for (int i = 0; i < body.length(); i++) {
-          Node* n = body.at(i);
-          if (get_pid_or_zero(n) == pid) {
-            tty->print("    ");
-            n->dump();
-          }
-        }
-      }
-#endif
-    }
-  }
-};
-
-// The C2 graph (specifically the memory graph), needs to be re-ordered.
-// (1) Build the PacksetGraph. It combines the dependency graph with the
-//     packset. The PacksetGraph gives us the dependencies that must be
-//     respected after scheduling.
-// (2) Schedule the PacksetGraph to the memops_schedule, which represents
-//     a linear order of all memops in the body. The order respects the
-//     dependencies of the PacksetGraph.
-// (3) If the PacksetGraph has cycles, we cannot schedule. Abort.
-// (4) Use the memops_schedule to re-order the memops in all slices.
-void SuperWord::schedule() {
```

Comment:
Note: closest new method is `SuperWord::vtransform`: build and schedule the graph, but then also apply it.